README.md

```diff
@@ -1 +1,83 @@
-## taskrunner
+# Taskrunner
+A configurable taskrunner written in Go. Taskrunner can streamline your build system by creating reusable task definitions with names, descriptions, dependencies, and a run function written in Go. Taskrunner comes with a shell interpreter, so it can run arbitrary functions on the command line.
+
+## Setup
+Add taskrunner as a dependency `go get -v github.com/samsarahq/taskrunner`, and create the following `main.go` file:
+```go
+package main
+
+import (
+	"github.com/samsarahq/taskrunner"
+	"github.com/samsarahq/taskrunner/clireporter"
+)
+
+func main() {
+	taskrunner.Run(clireporter.StdoutOption)
+}
+```
+
+## Run a task
+Taskrunner can be started by running `go run .`, it is possible to limit log output with `PRINT_LOG_LEVEL=error`. To run an individual task specify the task name `go run . my/task`. For a full list of tasks use `go run . -describe`, or for a list of names `go run . -list`. It may be a good idea to alias `taskrunner` to run taskrunner.
+
+## Example tasks
+Add a `tasks.go` file, this contains all task descriptions:
+```go
+package main
+
+import (
+	"context"
+
+	"github.com/samsarahq/taskrunner"
+	"github.com/samsarahq/taskrunner/goextensions"
+	"github.com/samsarahq/taskrunner/shell"
+)
+
+// Run a simple task.
+var myTask = taskrunner.Add(&taskrunner.Task{
+	Name:         "my/task",
+	Run: func(ctx context.Context, shellRun shell.ShellRun) error {
+		return shellRun(ctx, `echo Hello World`)
+	},
+})
+
+// Run a task depending on another task.
+var myDependentTask = taskrunner.Add(&taskrunner.Task{
+	Name:         "my/dependent/task",
+	Dependencies: []*taskrunner.Task{myTask},
+	Run: func(ctx context.Context, shellRun shell.ShellRun) error {
+		return shellRun(ctx, `echo Hello Again`)
+	},
+})
+
+
+// Run a task which monitors a file and reruns on changes
+// a change will also invalidate dependencies.
+var myGoTask = taskrunner.Add(&taskrunner.Task{
+	Name:         "my/go/task",
+	Description:  "Run a go file and re-run when it changes",
+	Run: func(ctx context.Context, shellRun shell.ShellRun) error {
+		return shellRun(ctx, `cd src/example && go run .`)
+	},
+	Sources: []string{"src/example/**/*.go"},
+})
+
+// Run a task using WrapWithGoBuild to automatically setup
+// invalidation for the task according to the impact graph
+// of the specified package.
+var builder = goextensions.NewGoBuilder()
+
+var myWrappedTask = taskrunner.Add(&taskrunner.Task{
+	Name: "my/wrapped/task",
+}, builder.WrapWithGoBuild("builtin"))
```

Comment:
can we change this to 
```golang
var builder = goextensions.NewGoBuilder()
var myWrappedTask = taskrunner.Add(&taskrunner.Task{
	Name: "my/wrapped/task",
        Run: func(ctx context.Context, shellRun shell.ShellRun) error {
		return shellRun(ctx, `example`)
	},
}, builder.WrapWithGoBuild("src/example"))
```

this should showcase the building and

Reply:
I was trying this out originally and I have probably made some sort of configuration issue but I always recieve this error message:
```
 building packages: src/example package src/example is not in GOROOT (/usr/local/Cellar/go/1.16.3/libexec/src/src/example)
 ```

I am wondering if this has anything to do with #42

Reply:
I have updated the readme regardless.

Reply:
i think this is OK, this jsut has to do with GOPATH configuration. not entirely sure what's going wrong without looking at your actual toy repo, but I think it's OK to merge

Reply:
my bad, it looks like this should be `builder.WrapWithGoBuild("example")` the src removed because it is double counting the src/src in `/usr/local/Cellar/go/1.16.3/libexec/src/src/example)`. 

I got this working with this example output:
```
$ go install taskrunner
$ taskrunner my/wrapped/task
2021/07/15 16:53:58 Using config /home/ubuntu/go/src/testtaskrunner/workspace.taskrunner.json
2021/07/15 16:53:58 Desired tasks: my/wrapped/task
2021/07/15 16:53:58 Watch mode: false
my/wrapped/task > Started
my/wrapped/task > building packages: exampledone building packages
my/wrapped/task > Example with go build!!
my/wrapped/task > Completed (0.58s)
```

contents of: `go/src/example/main.go`
```golang
 package main
 
 import "fmt"

func main() {
     fmt.Println("Example with go build!!")
}
```

contents of: `go/src/taskrunner/main.go`
```
package main

import (
	"github.com/samsarahq/taskrunner"
	"github.com/samsarahq/taskrunner/clireporter"
)

func main() {
	taskrunner.Run(clireporter.StdoutOption)
}

// Run a task using WrapWithGoBuild to automatically setup
// invalidation for the task according to the import graph
// of the specified package.
var builder = goextensions.NewGoBuilder()

var myWrappedTask = taskrunner.Add(&taskrunner.Task{
	Name: "my/wrapped/task",
        Run: func(ctx context.Context, shellRun shell.ShellRun) error {
		return shellRun(ctx, `example`)
	},
}, builder.WrapWithGoBuild("example"))
```

This also assumes that `go/bin` is in the users `$PATH`